lib/blacklight/search_state.rb

```diff
@@ -14,30 +14,67 @@ class SearchState
 
     delegate :facet_configuration_for_field, to: :blacklight_config
 
+    def self.modifiable_params(params)
+      if params.respond_to?(:to_unsafe_h)
+        # This is the typical (not-ActionView::TestCase) code path.
+        params = params.to_unsafe_h
+        # In Rails 5 to_unsafe_h returns a HashWithIndifferentAccess, in Rails 4 it returns Hash
+        params = params.with_indifferent_access if params.instance_of? Hash
+      elsif params.is_a? Hash
```

Comment:
I think this clause was already removed from the main branch, so we don't want to reintroduce it.

Reply:
Thanks for catching that! Added a commit.